### PR TITLE
[refactor] Upgrade swapper #347

### LINKED ISSUE #347: [refactor] Upgrade swapper and replace swappable_dependency_latest
Replace these lines:

https://github.com/openwisp/openwisp-radius/pull/328/files#diff-353cd5338ca726ac57e356a8878f8ece02df54d683899b0c672803eb8f64b4e3R151-R152

With https://github.com/openwisp/django-swappable-models/commit/99b82971da494c624ae8bda3612d53bf5b3b02af

We'll need to increase the version of swappable to ~=1.2.0 here and in openwisp-users.

### PR DESCRIPTION
**Pull request by yashikajotwani12**
Fixes #347

### CODE REVIEW

File: openwisp_radius/migrations/__init__.py [hunk 1]
```diff
@@ -147,6 +149,43 @@ def populate_phonetoken_phone_number(apps, schema_editor):
...
+def get_model_name(app_label, model):
+    return is_swapped(app_label, model) or join(app_label, model)
 
-def swappable_dependency_latest(value):
-    return (swappable_dependency(value)[0], '__latest__')
+def dependency(app_label, model, latest=True):
+    dependencies = swappable_dependency(get_model_name(app_label, model))
+    if not latest:
+        return dependencies
+    return dependencies[0], '__latest__'
```
Comment (codesankalp) [inline openwisp_radius/migrations/__init__.py:191]: You don't have to repeat the whole code of swapper again, see my suggestions:

```suggestion
    def swappable_dependency_latest(value):
        return dependency(*value.split('.'), latest=True)
```

File: openwisp_radius/migrations/__init__.py [hunk 2]
```diff
@@ -147,6 +149,43 @@ def populate_phonetoken_phone_number(apps, schema_editor):
...
+        return False
+
+def join(app_label, model):
+    return "{app_label}.{model}".format(
+        app_label=app_label,
+        model=model,
+    )
+
+def get_model_name(app_label, model):
+    return is_swapped(app_label, model) or join(app_label, model)
```
Comment (codesankalp) [inline openwisp_radius/migrations/__init__.py:185]:
```suggestion
```

File: openwisp_radius/migrations/__init__.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
 import swapper
+from swapper import dependency
 from django.conf import settings
 from django.contrib.auth.management import create_permissions
 from django.contrib.auth.models import Permission
 from django.db.migrations import swappable_dependency
 
 from ..utils import create_default_groups
+_prefixes = {}
```
Comment (codesankalp) [inline openwisp_radius/migrations/__init__.py:11]:
```suggestion
```

**Review by codesankalp — CHANGES_REQUESTED** [review]
@yashikajotwani12 Many changes are not required see my suggestions below.
Also, update the swapper version:
https://github.com/openwisp/openwisp-radius/blob/b19de59455d40484674a2be870f99a2ee62568fe/setup.py#L39

**Comment by yashikajotwani12** [comment]
> @yashikajotwani12 Many changes are not required see my suggestions below. Also, update the swapper version:
> 
> https://github.com/openwisp/openwisp-radius/blob/b19de59455d40484674a2be870f99a2ee62568fe/setup.py#L39

Thanks! I would update these changes.

File: openwisp_radius/migrations/__init__.py [hunk 4]
```diff
@@ -147,6 +147,5 @@ def populate_phonetoken_phone_number(apps, schema_editor):
         phone_token.phone_number = phone_token.user.phone_number
         phone_token.save(update_fields=['phone_number'])
 
-
 def swappable_dependency_latest(value):
-    return (swappable_dependency(value)[0], '__latest__')
+        return dependency(*value.split('.'), latest=True)
```
Comment (codesankalp) [inline openwisp_radius/migrations/__init__.py:151]:
```suggestion
    return dependency(*value.split('.'), latest=True)
 
```

**Review by codesankalp — CHANGES_REQUESTED** [review]
@yashikajotwani12 Please test this locally i.e. `./run-qa-checks` and `./runtests` before pushing.

File: openwisp_radius/migrations/__init__.py [hunk 5]
```diff
@@ -147,6 +147,5 @@ def populate_phonetoken_phone_number(apps, schema_editor):
         phone_token.phone_number = phone_token.user.phone_number
         phone_token.save(update_fields=['phone_number'])
 
-
 def swappable_dependency_latest(value):
-    return (swappable_dependency(value)[0], '__latest__')
+    return dependency(*value.split('.'), latest=True)
```
Comment (pandafy) [inline openwisp_radius/migrations/__init__.py:151]: I don't think we require this method anymore.

**Review by pandafy — CHANGES_REQUESTED** [review]
Following changes are required: 

```diff
diff --git a/openwisp_radius/migrations/0002_initial_openwisp_radius.py b/openwisp_radius/migrations/0002_initial_openwisp_radius.py
index 1cbfbbc..3f3e998 100644
--- a/openwisp_radius/migrations/0002_initial_openwisp_radius.py
+++ b/openwisp_radius/migrations/0002_initial_openwisp_radius.py
@@ -7,6 +7,7 @@ import django.core.validators
 import django.db.models.deletion
 import django.utils.timezone
 import model_utils.fields
+import swapper
 from django.conf import settings
 from django.db import migrations, models
 from swapper import get_model_name
@@ -14,10 +15,7 @@ from swapper import get_model_name
 import openwisp_users.mixins
 import openwisp_utils.base
 import openwisp_utils.utils
-from openwisp_radius.migrations import (
-    add_default_organization,
-    swappable_dependency_latest,
-)
+from openwisp_radius.migrations import add_default_organization
 
 
 class Migration(migrations.Migration):
@@ -31,7 +29,7 @@ class Migration(migrations.Migration):
     """
 
     dependencies = [
-        swappable_dependency_latest(settings.AUTH_USER_MODEL),
+        swapper.dependency(*settings.AUTH_USER_MODEL.split('.'), latest=True),
         ('openwisp_radius', '0001_initial_freeradius'),
     ]
 
diff --git a/openwisp_radius/migrations/__init__.py b/openwisp_radius/migrations/__init__.py
index 61fd7bd..bb9335a 100644
--- a/openwisp_radius/migrations/__init__.py
+++ b/openwisp_radius/migrations/__init__.py
@@ -146,7 +146,3 @@ def populate_phonetoken_phone_number(apps, schema_editor):
     for phone_token in PhoneToken.objects.all():
         phone_token.phone_number = phone_token.user.phone_number
         phone_token.save(update_fields=['phone_number'])
-
-
-def swappable_dependency_latest(value):
-    return (swappable_dependency(value)[0], '__latest__')
- ```

File: openwisp_radius/migrations/__init__.py (hunk 4 above)
Comment (pandafy) [reply openwisp_radius/migrations/__init__.py]: Do we still need this method?

**Comment by codesankalp** [comment]
We can also use `swapper.split`

https://github.com/openwisp/django-swappable-models/blob/0f84196b65d316895d33a4eef28c7db86cb93ff5/swapper/__init__.py#L99-L101

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
I think this line has to be updated to point to the new version:
https://github.com/openwisp/openwisp-radius/blob/master/setup.py#L39

The rest looks good.

@codesankalp what about improving `swapper.dependency` to allow passing a string which is split automatically like doing here (to avoid repeating this code in every module)?

**Comment by yashikajotwani12** [comment]
> I think this line has to be updated to point to the new version: https://github.com/openwisp/openwisp-radius/blob/master/setup.py#L39
> 
> The rest looks good.
> 
I have updated this.

**Review by nemesisdesign — COMMENTED** [review]
Looks good, I did one final improvement: https://github.com/openwisp/openwisp-radius/pull/352/commits/4586892295d53b1af87bc42599d55a8027251339.
Waiting for the build to finish before merging.

**Review by nemesisdesign — APPROVED** [review]
Thanks @yashikajotwani12 @codesankalp

**Pull request merged**